### PR TITLE
Move iMIP to its own interface

### PR DESCRIPTION
**Pull request by miaulalala**
and clean up the code a bit.

The old implementation forced every writable calendar to support iMIP.

Added:
- author

Removed:
- unused imports

Changed:
- moved iMIP handling to new interface
- pointed code to new implementation

Documentation: https://github.com/nextcloud/documentation/pull/9269

### CODE REVIEW

File: lib/public/Calendar/IHandleImipMessage.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+use OCP\Calendar\Exceptions\CalendarException;
+
+/**
+ * Extends the current ICalendar interface
+ * to add a public write method to handle
+ * iMIP data
+ *
+ * @link https://www.rfc-editor.org/rfc/rfc6047
+ *
+ * @since 25.0.0
```
Comment (tcitworld) [inline lib/public/Calendar/IHandleImipMessage.php:36]: Are we rewriting history or not? :)

Comment (miaulalala) [reply lib/public/Calendar/IHandleImipMessage.php:36]: hehe I wasn't sure. @ChristophWurst your opinion?

Comment (ChristophWurst) [reply lib/public/Calendar/IHandleImipMessage.php:36]: We won't backport this API change

**Comment by miaulalala** [comment]
Should the IManager's methods also be extracted? Same issue here, all implementations of IManager now need to support iMIP.

**Comment by ChristophWurst** [comment]
> Should the IManager's methods also be extracted? Same issue here, all implementations of IManager now need to support iMIP.

IManager is a consumed/injected service. The implementation is done by the server. This is fine.

The breaking change only occurs when you add methods to interfaces that are implemented by apps.

File: lib/public/Calendar/IHandleImipMessage.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ * iMIP data
+ *
+ * @link https://www.rfc-editor.org/rfc/rfc6047
+ *
+ * @since 26.0.0
+ */
+interface IHandleImipMessage extends ICalendar {
+
+	/**
+	 * @since 26.0.0
```
Comment (ChristophWurst) [inline lib/public/Calendar/IHandleImipMessage.php:41]: Should have a description

File: lib/public/Calendar/IHandleImipMessage.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	 *
+	 * @throws CalendarException
```
Comment (ChristophWurst) [inline lib/public/Calendar/IHandleImipMessage.php:43]: Ideally document *under which conditions* this could be thrown

**Review by ChristophWurst — APPROVED** [review]

**Comment by miaulalala** [comment]
Documentation PR here: https://github.com/nextcloud/documentation/pull/9269

**Review by ChristophWurst — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]

**Review by tcitworld — APPROVED** [review]

**Comment by miaulalala** [comment]
Drone failure unrelated

**Comment by juliushaertl** [comment]
PR to fix in https://github.com/nextcloud/server/pull/34954

**Pull request merged**